src/core/js/models/adaptModel.js

```diff
@@ -719,6 +719,8 @@ define([
                 var child = children[i];
 
                 child.set("_isLocked", this.shouldLock(child));
+
+                if (child.get('_type') === 'menu') child.setCustomLocking();
```

Comment:
I've had another thought on this today (apologies).

### Assumption
Works for only course or menu contentObjects with child menus.
The children will have custom locking if the parent has custom locking.

### Issue
The child menu might have an alternative locking type.

### Suggestion
Switch from `child.setCustomLocking();` to `child.checkLocking();`

```suggestion
                if (child.get('_type') === 'menu') {
                    child.checkLocking();
                }
```